packages/client-sdk-nodejs/src/internal/vector-index-data-client.ts

```diff
@@ -212,9 +270,35 @@ export class VectorIndexDataClient implements IVectorIndexDataClient {
                   id: hit.id,
                   distance: hit.distance,
                   metadata: hit.metadata.reduce((acc, metadata) => {
-                    acc[metadata.field] = metadata.string_value;
+                    const field = metadata.field;
+                    switch (metadata.value) {
+                      case 'string_value':
+                        acc[field] = metadata.string_value;
+                        break;
+                      case 'integer_value':
+                        acc[field] = metadata.integer_value;
+                        break;
+                      case 'double_value':
+                        acc[field] = metadata.double_value;
+                        break;
+                      case 'boolean_value':
+                        acc[field] = metadata.boolean_value;
+                        break;
+                      case 'list_of_strings_value':
+                        acc[field] = metadata.list_of_strings_value.values;
+                        break;
+                      default:
+                        resolve(
+                          new VectorSearch.Error(
+                            new UnknownError(
+                              'Search responded with an unknown result'
+                            )
+                          )
+                        );
+                        break;
+                    }
                     return acc;
-                  }, {} as Record<string, string>),
+                  }, {} as Record<string, string | number | boolean | Array<string>>),
```

Comment:
just curious if you tried a type alias for this in JS too, and observed the same issue where the IDEs didn't expand it in a helpful way

Reply:
Same thing in the sense that if we make a type alias `T` that is used in a function signature for `foo`, then if you type `foo` you will only see `T`, unless you manually click the function, go to definition, and then hover over `T`.